main.go

```diff
@@ -118,11 +125,30 @@ func prefetch(url string) (string, error) {
 	contents, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		log.Print("Prefetch failed reading body for: ", url, err)
+	}
+	dname, err := os.MkdirTemp("", "")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp dir: %w", err)
+	}
```

Comment:
Just noticed that this directory should be removed after use here, and not further down below, i.e.

```suggestion
	dname, err := os.MkdirTemp("", "")
	if err != nil {
		return "", fmt.Errorf("failed to create temp dir: %w", err)
	}
	defer func() {
		if err := os.RemoveAll(dname); err != nil {
			log.Printf("failed to clean up temp dir %q: %v", dname, err)
		}
	}()
```